runtime/hydradx/src/xcm.rs

```diff
@@ -306,8 +309,41 @@ pub type LocationToAccountId = (
 	SiblingParachainConvertsVia<Sibling, AccountId>,
 	// Straight up local `AccountId32` origins just alias directly to `AccountId`.
 	AccountId32Aliases<RelayNetwork, AccountId>,
+	// Generate remote accounts according to polkadot standards
+	xcm_account_derivation::HashedDescriptionDescribeFamilyAllTerminal<AccountId>,
+	// Convert ETH to local substrate account
+	EvmAddressConversion<RelayNetwork>,
 );
 
+//	,
+/// Converts Account20 (ethereum) addresses to AccountId32 (substrate) addresses.
+pub struct EvmAddressConversion<Network>(PhantomData<Network>);
+impl<Network: Get<Option<NetworkId>>> xcm_executor::traits::Convert<MultiLocation, AccountId>
+	for EvmAddressConversion<Network>
+{
+	fn convert(location: MultiLocation) -> Result<AccountId, MultiLocation> {
+		match location {
+			MultiLocation {
+				parents: 0,
+				interior: X1(AccountKey20 { network: _, key }),
+			} => {
+				let account_32 = ExtendedAddressMapping::into_account_id(H160::from(key));
+				Ok(account_32)
+			}
+			_ => Err(location),
+		}
+	}
+
+	//TODO: fix this reverse
+	fn reverse(who: AccountId) -> Result<MultiLocation, AccountId> {
```

Comment:
wondering if this is correct, of we should just throw Err in this case

Reply:
iirc according to @mrq1911 we should use last 20 bytes from 32-byte address to determine it